packages/next/server/base-server.ts

```diff
@@ -219,6 +216,12 @@ export default abstract class Server {
   public readonly hostname?: string
   public readonly port?: number
 
+  protected abstract getHasStaticDir(): boolean
+  protected abstract getPagesManifest(): PagesManifest | undefined
+  protected abstract getBuildId(): string
+  protected abstract generatePublicRoutes(): Route[]
+  protected abstract getFilesystemPaths(): Set<string>
```

Comment:
What do we think about trying to make these async when possible? I know that we can have some of this pre-calculated, but I feel like if it's easy during rewrite, making them async is just more flexible later on

Reply:
Yeah I think we can totally do that in an upcoming PR. Would be good to do the same for `require` too!